students/BrianNisonger/Lesson5/assignment/database.py

```diff
@@ -0,0 +1,108 @@
+import os
+import csv
+import logging
+from pymongo import MongoClient
+
+
+class MongoDBConnection(object):
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def calculate_availability(product_list, rental_list):
+    for product in product_list:
+        count = 0
+        for rental in rental_list:
+            if rental["product_id"] == product["product_id"]:
+                count += 1
+        product_available = int(product["quantity_available"]) - count
+        product["quantity_available"] = product_available
+    return product_list
+
+
+def make_product_dict(prod_available_list):
+    fields = ["description", "product_type", "quantity_available"]
+    product_values_dict = {}
+    product_dict = {}
+    for product in prod_available_list:
+        product_values_dict = {}
+        for field in fields:
+            product_values_dict[field] = product[field]
+        if product["quantity_available"] > 0:
+            product_dict[product["product_id"]] = product_values_dict
+    return product_dict
+
+
+def return_user_ids(rental_list):
+    return [rental["user_id"] for rental in rental_list]
+
+
+def make_customer_dict(user_list):
+    fields = ["name", "address", "phone_number", "email"]
+    customer_values_dict = {}
+    customer_dict = {}
+    for customer in user_list:
+        customer_values_dict = {}
+        for field in fields:
+            customer_values_dict[field] = customer[field]
+            customer_dict[customer["user_id"]] = customer_values_dict
```

Comment:
I think you don't want this line:
```
customer_dict[customer["user_id"]] = customer_values_dict
```
inside your second for loop (which is also a candidate for a comprehension), because you are adding the customer_values_dict every field instead of after the dictionary is fully populated.